### ISSUE TITLE
[FEATURE REQUEST] : Add support to Altair charts in Python

### ISSUE THREAD

**Issue by djouallah** [rstudio/rstudio#4259]
Build 1296 (d5eaed87)
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.0 Chrome/69.0.3497.128 Safari/537.36

it will be very helpful, if Rstudio support natively the output of Altair charts, right now the code works, but instead of generate a chart, it show only a json description

```
import altair as alt
floor_Plan = alt.topo_feature('https://raw.githubusercontent.com/djouallah/R_leaflet_MAP/master/foundation.json', 'foundation')
source = 'https://raw.githubusercontent.com/djouallah/R_leaflet_MAP/master/data/data.csv'
map = alt.Chart(floor_Plan).mark_geoshape().encode(
color='progress:Q',
tooltip=['properties.id:N','progress:N']).transform_lookup(
    lookup='properties.id',
    from_=alt.LookupData(source, 'id', ['progress']))

map
```
R support already the render of that specification using https://github.com/vegawidget/vegawidget  but it will be really nice if it is supported natively in Python

**Comment by jmcphers** [comment]
Agreed that it'd be very nice! However this has been open for over a year without any additional votes or attention so unlikely we'll get to it soon; closing for now.

**Comment by joelostblom** [comment]
I would love to see Altair support! @jmcphers Would you consider reopening this given that there are a few thumbs up now and also two related issues with several people expressing interested?

It seems like the foundation for this already exists in vegawidget, for example something like this works in a notebook inside RStudio, but not when knitting:

````
---
title: "Altair-reticulate"
output: html_document
---

```{r load_packages}
library(vegawidget) # Need to install this using: install.packages('vegawidget')
library(reticulate)
# use_python('/usr/local/bin/python3') # you may not need this, comment out if so
```

```{python}
import altair as alt
import pandas as pd
source = pd.DataFrame({
    'a': ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I'],
    'b': [28, 55, 43, 91, 81, 53, 19, 87, 52]
})
chart1 = alt.Chart(source).mark_bar().encode(
    alt.X('a:N',title='X-Axis label'),
    alt.Y('b:Q',title='Y-axis label')).to_json()
chart1
```

```{r}
as_vegaspec(py$chart1)
```
````

**Comment by kevinushey** [comment]
See also: https://github.com/rstudio/reticulate/issues/711

**Comment by cderv** [comment]
> It seems like the foundation for this already exists in vegawidget, for example something like this works in a notebook inside RStudio, but not when knitting

About the knitting issue, it is with R 4+ only and it is something related to **vegawiget** not registering the S3 method correctly. (https://github.com/vegawidget/vegawidget/issues/132) 
If the registration is done properly, this method works - using python chunk then using R to display.
